compiler/rustc_mir_transform/src/const_prop.rs

```diff
@@ -386,7 +388,17 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 op
             }
             Err(e) => {
-                trace!("get_const failed: {:?}", e.into_kind().debug());
+                let msg = ty::tls::with(move |tcx| {
+                    let handler = &tcx.sess.parse_sess.span_diagnostic;
+                    let err = InterpErrorExt(e.into_kind());
+                    let builder = err.into_diagnostic(handler);
```

Comment:
@fee1-dead I am not sure how to go from `DiagnosticBuilder` to `String`.

Reply:
This is what I did in miri:

https://github.com/fee1-dead-contrib/rust/blob/f6c2bc5c24fd08200c0e4438d981ca58dc71f488/src/tools/miri/src/diagnostics.rs#L333-L343

at this point we could maybe extract this as a method and just stress that this shouldn't be used for user-facing diagnostics (just debug messages) in the docs.